src/VisualStudio/Core/Test.Next/Services/ServiceHubServicesTests.cs

```diff
@@ -669,6 +685,17 @@ public async Task InProcAndRemoteWorkspaceAgree20(bool syncWithRemoteServer)
                 ImmutableArray.Create(("SG2.cs", CreateText(contents)), ("SG1.cs", CreateText(contents))));
         }
 
+        [Theory, CombinatorialData]
+        public async Task InProcAndRemoteWorkspaceAgree21(
+            bool syncWithRemoteServer, bool useBOM1, bool useMemoryStream1, bool useBOM2, bool useMemoryStream2)
+        {
+            var contents = Guid.NewGuid().ToString();
+            await TestInProcAndRemoteWorkspace(
+                syncWithRemoteServer,
+                ImmutableArray.Create(("SG.cs", CreateStreamText(contents, useBOM: useBOM1, useMemoryStream: useMemoryStream1))),
+                ImmutableArray.Create(("SG.cs", CreateStreamText(contents, useBOM: useBOM2, useMemoryStream: useMemoryStream2))));
```

Comment:
was able to hit the issue easily with these variying combinations of bom/no-bom.